### ISSUE TITLE
don't require username when certificate based authentication is used

### ISSUE THREAD

**Issue by ghost** [karimra/gnmic#641]
Some servers can extract usernames from certificates. This may be as simple as extracting the common name but here is also the [SPIFFE-ID ](https://spiffe.io/docs/latest/spiffe-about/spiffe-concepts/) which was made specifically for this purpose.

Enforcing that the user provides a username through metadata when using certificate based authentication leaves it up to the server to decide which one it should use, with some servers returning an error due to this ambiguous behaviour.

**Comment by karimra** [comment]
Agree, it makes sense to not require a username/password when running a command.
